### PR TITLE
Support querying issues and incidents with kcidb(-db)-query.

### LINKED ISSUE #398: Support querying issues and incidents with kcidb(-db)-query
At the moment, both the `kcidb-query` and its twin `kcidb-db-query` only support specifying the IDs for checkouts, builds, and tests to retrieve. However, the latest I/O and database schema versions have added support for incidents and issues. Add support for specifying IDs of those new objects to both tools, by adding support for two more option pairs: `-i`/`--issue-id` and `-n`/`--incident-id`. This involves modifying the `kcidb.db.QueryArgumentParser` class.

### PR DESCRIPTION
**Pull request by octonawish-akcodes**
This PR closes #398 
@spbnick need reviews

### CODE REVIEW

File: kcidb/db/__init__.py
Comment (spbnick) [inline kcidb/db/__init__.py:556]: Also, could you switch the order of those? "Issues" are parents of "incidents" in the schema, and the rest of these options are sorted by schema depth.

Comment (octonawish-akcodes) [reply kcidb/db/__init__.py:556]: okay

**Comment by octonawish-akcodes** [comment]
Check now sir

**Comment by spbnick** [comment]
Nicely done, thank you, @octonawish-akcodes. Please rebase on the latest changes in main (database tests were rewritten by @NikkyXO), and let's get this merged.

**Comment by octonawish-akcodes** [comment]
> Nicely done, thank you, @octonawish-akcodes. Please rebase on the latest changes in main (database tests were rewritten by @NikkyXO), and let's get this merged.

Some tests are now failing because of the latest change in rewritten of tests, so I am currently figuring out the support.

**Comment by spbnick** [comment]
I'm [reverting](https://github.com/kernelci/kcidb/pull/419) the test change that was interfering with this PR, because it did break some tests indeed. We should be able to merge this as is, once that is merged.

**Comment by spbnick** [comment]
No. Please stand by for the merge to happen, after CI passes.

You would've needed to rebase and redo your test changes if that commit worked.

**Review by spbnick — APPROVED** [review]
LGTM! Thank you, @octonawish-akcodes. Merging!

**Pull request merged**
